x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/action/saml/TransportSamlInvalidateSessionAction.java

```diff
@@ -109,23 +109,24 @@ private void findAndInvalidateTokens(SamlRealm realm, SamlLogoutRequestHandler.R
             return;
         }
 
-        tokenService.findActiveTokensForRealm(realm.name(), containsMetadata(tokenMetadata), ActionListener.wrap(tokens -> {
-            LOGGER.debug("Found [{}] token pairs to invalidate for SAML metadata [{}]", tokens.size(), tokenMetadata);
-            if (tokens.isEmpty()) {
-                listener.onResponse(0);
-            } else {
-                tokenService.invalidateAllTokens(tokens, ActionListener.wrap(tokensInvalidationResult -> {
-                    if (LOGGER.isInfoEnabled() && tokensInvalidationResult.getErrors().isEmpty() == false) {
-                        try (XContentBuilder builder = XContentBuilder.builder(XContentType.JSON.xContent())) {
-                            tokensInvalidationResult.toXContent(builder, ToXContent.EMPTY_PARAMS);
-                            LOGGER.info("Failed to invalidate some SAML access or refresh tokens {}", Strings.toString(builder));
-                        }
+        tokenService.invalidateActiveTokens(
```

Comment:
Reused the `tokenService#invalidateActiveTokens` method in order to avoid exposing the token search method (previously the `findActiveTokensForRealm` method), which now is private.